src/python_bindings/fastvdf.cpp

```diff
@@ -11,12 +11,15 @@ PYBIND11_MODULE(chiavdf, m) {
     // Creates discriminant.
     m.def("create_discriminant", [] (const py::bytes& challenge_hash, int discriminant_size_bits) {
         std::string challenge_hash_str(challenge_hash);
-        py::gil_scoped_release release;
-        auto challenge_hash_bits = std::vector<uint8_t>(challenge_hash_str.begin(), challenge_hash_str.end());
-        integer D = CreateDiscriminant(
-            challenge_hash_bits,
-            discriminant_size_bits
-        );
+        integer D;
+        {
+            py::gil_scoped_release release;
+            auto challenge_hash_bits = std::vector<uint8_t>(challenge_hash_str.begin(), challenge_hash_str.end());
+            D = CreateDiscriminant(
+                challenge_hash_bits,
+                discriminant_size_bits
+            );
+        }
         return D.to_string();
```

Comment:
I don't think you need to hold the GIL during the call to `to_string()`. That's the only change here, right? you acquire the lock a bit earlier

Reply:
I wanted to make sure i had a copy before doing the gil_scoped_release, basically adding some more brackets